protocol/x/accountplus/testutils/testutils.go

```diff
@@ -0,0 +1,52 @@
+package testutils
+
+import (
+	"github.com/dydxprotocol/v4-chain/protocol/x/accountplus/types"
+	"github.com/google/go-cmp/cmp"
+)
+
+func CompareTimestampNonceDetails(actualDetails, expectedDetails *types.TimestampNonceDetails) bool {
```

Comment:
Why do we need this special compare function? And do we still need it if it's no longer a pointer?

Reply:
Not too familiar with Go but my understanding is that TimestampNonceDetails is a struct which contains a slice so comparison is not directly supported. Moreover, slice comparison in Go is by reference. 

Many unit tests will come down to comparing TimestampNonceDetails so I thought it would be helpful to create a helper function (same applies for AccountState).

As for if theres are not pointers, i think the above issues are still relevant.

Reply:
> TimestampNonceDetails is a struct which contains a slice so comparison is not directly supported.

IIRC `require.Equal` uses `reflect.DeepEqual` so it should be supported. Otherwise we would've needed to write a lot of recursive compare utilities